File: sdk/python/src/agentpay_sdk/obligation.py
"""Obligation queries (mirrors sdk/typescript/src/obligation.ts)."""

from .client import AgentPayClient
from .types import ObligationResponse


def get_obligation(client: AgentPayClient, obligation_id: str) -> ObligationResponse:
    """Get an obligation by ID.

    GET /v1/rails/obligations/:id
    """
    data = client.get(f"/v1/rails/obligations/{obligation_id}")
    return ObligationResponse(
        obligation_id=data["obligationId"],
        sla_id=data["slaId"],
        consumer_handle=data["consumerHandle"],
        provider_handle=data["providerHandle"],
        consumer_smart_account=data["consumerSmartAccount"],
        provider_smart_account=data["providerSmartAccount"],
        amount_usdc_micro=data["amountUsdcMicro"],
        asset=data["asset"],
        network=data["network"],
        nonce=data["nonce"],
        finality_state=data["finalityState"],
        policy_decision_id=data["policyDecisionId"],
        tx_hash=data.get("txHash"),
        evidence_hash=data.get("evidenceHash"),
        created_at=data["createdAt"],
    )

File: sdk/typescript/src/obligation.ts
import type { AgentPayClient } from "./client";
import type { ObligationResponse, SettleResponse } from "./types";

/**
 * Get an obligation by ID (R11.1).
 *
 * GET /v1/rails/obligations/:id
 *
 * @param client - The AgentPay API client
 * @param obligationId - UUIDv7 obligation identifier
 * @returns The full obligation record
 */
export async function getObligation(
  client: AgentPayClient,
  obligationId: string,
): Promise<ObligationResponse> {
  return client.get<ObligationResponse>(`/v1/rails/obligations/${obligationId}`);
}
